### PR TITLE
Fix Transcript Download Failure During Post Chat When User Ends Conversation

### PR DESCRIPTION
**Pull request by xTEddie**
### Description
- Fix issue where `Transcript Download` would not working during `PostChatSurvey` after user ended the conversation

## Solution Proposed
- `liveChatContext` needed to be retrieve from `inMemoryState`

### Acceptance criteria
_Define what are the conditions to consider the PR has achieved the intended goal_

## Test cases and evidence
_Include what tests cases were considered, any evidence of testing for future references, to identify any corner cases, etc_

### Sanity Tests
-   [ ] You have tested all changes in Popout mode
-   [ ] You have tested all changes in cross browsers i.e Edge, Chrome, Firefox, Safari and mobile devices(iOS and Android)
-   [ ] Your changes are included in the [CHANGELOG](../CHANGE_LOG.md)

## A11y
- [ ] You have run the [Automated Accessibility Check](https://accessibilityinsights.io/docs/en/windows/getstarted/automatedchecks) and have no reported issues

__*Please provide justification if any of the validations has been skipped.*__

### CODE REVIEW

File: chat-widget/src/components/footerstateful/downloadtranscriptstateful/DownloadTranscriptStateful.tsx [hunk 1]
```diff
@@ -167,8 +169,14 @@ const beautifyChatTranscripts = (chatTranscripts: string, renderMarkDown?: (tran
 
 // eslint-disable-next-line @typescript-eslint/no-explicit-any
 export const downloadTranscript = async (chatSDK: any, downloadTranscriptProps: IDownloadTranscriptProps, state?: ILiveChatWidgetContext) => {
-    // Need to keep existing live chat context for scenarios when trnascript is downloaded after endchat
-    const liveChatContext = state?.domainStates?.liveChatContext;
+    const inMemoryState = executeReducer(state as ILiveChatWidgetContext, { type: LiveChatWidgetActionType.GET_IN_MEMORY_STATE, payload: null });
+
+    // Need to keep existing live chat context for scenarios when transcript is downloaded after endchat
```
Comment (elopezanaya) [inline chat-widget/src/components/footerstateful/downloadtranscriptstateful/DownloadTranscriptStateful.tsx:174]: why not to return directly from inMemory if is not null?

**Review by elopezanaya — CHANGES_REQUESTED** [review]

File: chat-widget/src/components/footerstateful/downloadtranscriptstateful/DownloadTranscriptStateful.tsx (hunk 1 above)
Comment (xTEddie) [reply chat-widget/src/components/footerstateful/downloadtranscriptstateful/DownloadTranscriptStateful.tsx:174]: It's because the issue only happens specifically on `Post Chat Survey Link Mode` where `liveChatContext` is undefined in the `state`. If  `liveChatContext` is not defined from the state, then we'd try it from the `inMemoryState`

**Review by elopezanaya — APPROVED** [review]

**Review by kevinlin98z — APPROVED** [review]
